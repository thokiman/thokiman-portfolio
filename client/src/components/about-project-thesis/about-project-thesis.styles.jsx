import styled, { css } from "styled-components";

export const AboutThesisProjectContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 95%;
  margin: 0 auto;
`;

export const AboutThesisProjectTitle = styled.div`
  font-size: 400%;
  padding: 10px;
  color: #e3e1e4;
  letter-spacing: 0.04em;
  text-align: left;
  display: flex;
  align-items: center;
  @media only screen and (max-width: 800px) {
    font-size: 350%;
  }
`;

export const AboutThesisProjectAttributes = styled.div`
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-auto-rows: repeat(5, 1fr);
  gap: 2px 2px;
  grid-template-areas:
    "name-thesis-project name-thesis-project name-thesis-project name-thesis-project name-thesis-project"
    "type-thesis-project type-thesis-project start-end-date-thesis-project start-end-date-thesis-project start-end-date-thesis-project"
    "type-thesis-project type-thesis-project length-thesis-project length-thesis-project length-thesis-project"
    "location-thesis-project location-thesis-project description-thesis-project description-thesis-project description-thesis-project"
    "location-thesis-project location-thesis-project description-thesis-project description-thesis-project description-thesis-project";

  border-bottom: #74c6d4 solid 1px;
  margin: 10px auto;
`;

const AboutThesisProjectStyles = css`
  width: 100%;
  margin: 10px auto;
  text-align: left;
  display: flex;
  align-items: center;
  text-align: left;
  letter-spacing: 0.04em;
  color: #e3e1e4;
`;
export const AboutNameThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: name-thesis-project;
  font-size: 220%;
  padding: 10px;
  @media only screen and (max-width: 800px) {
    font-size: 190%;
  }
`;

export const AboutTypeThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: type-thesis-project;
  font-size: 150%;
  line-height: 1.1em;
  border-bottom: #74c6d4 solid 1px;
`;

export const AboutLocationThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: location-thesis-project;
  font-size: 150%;
  line-height: 1.1em;
`;
export const AboutStartEndDateThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: start-end-date-thesis-project;
  font-size: 130%;
  border-bottom: #74c6d4 solid 1px;
  line-height: 1.1em;
  letter-spacing: 0.05em;
  padding: 10px 0;
  @media only screen and (max-width: 800px) {
    font-size: 110%;
  }
`;
export const AboutLengthThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: length-thesis-project;
  font-size: 130%;
  line-height: 1.1em;
  letter-spacing: 0.05em;
  padding: 10px 0;
  border-bottom: #74c6d4 solid 1px;
  @media only screen and (max-width: 800px) {
    font-size: 110%;
  }
`;
export const AboutDescriptionThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: description-thesis-project;
  font-size: 120%;
  line-height: 1.1em;
  letter-spacing: 0.05em;
  @media only screen and (max-width: 800px) {
    font-size: 100%;
  }
`;

AboutThesisProjectContainer.displayName = "AboutThesisProjectContainer";
AboutThesisProjectTitle.displayName = "AboutThesisProjectTitle";
AboutThesisProjectAttributes.displayName = "AboutThesisProjectAttributes";
AboutNameThesisProject.displayName = "AboutNameThesisProject";
AboutTypeThesisProject.displayName = "AboutTypeThesisProject";
AboutLocationThesisProject.displayName = "AboutLocationThesisProject";
AboutStartEndDateThesisProject.displayName = "AboutStartEndDateThesisProject";
AboutLengthThesisProject.displayName = "AboutLengthThesisProject";
AboutDescriptionThesisProject.displayName = "AboutDescriptionThesisProject";
