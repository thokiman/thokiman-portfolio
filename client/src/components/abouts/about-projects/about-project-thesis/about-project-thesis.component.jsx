import React from 'react';
import {
  AboutDescriptionThesisProject,
  AboutLengthThesisProject,
  AboutLocationThesisProject,
  AboutNameThesisProject,
  AboutStartEndDateThesisProject,
  AboutThesisProjectAttributes,
  AboutThesisProjectContainer,
  AboutThesisProjectTitle,
  AboutTypeThesisProject,
} from './about-project-thesis.styles';

export const AboutThesisProject = ({ thesis }) => {
  return (
    <AboutThesisProjectContainer className='about-thesis-project-content'>
      <AboutThesisProjectTitle className='about-thesis-project-title'>
        Thesis
      </AboutThesisProjectTitle>
      {thesis.map(
        ({
          id,
          name,
          type,
          startDate,
          endDate,
          length,
          location,
          description,
        }) => {
          return (
            <AboutThesisProjectAttributes
              className='about-thesis-project-attributes'
              key={id}
            >
              <AboutNameThesisProject className='about-thesis-project'>
                {name}
              </AboutNameThesisProject>
              <AboutTypeThesisProject className='about-thesis-project'>
                {type.charAt(0).toUpperCase()}
                {type.slice(1, 7)}
                {type.charAt(7).toUpperCase()}
                {type.slice(8)}
              </AboutTypeThesisProject>
              <AboutStartEndDateThesisProject className='about-thesis-project'>
                {startDate} - {endDate}
              </AboutStartEndDateThesisProject>
              <AboutLengthThesisProject className='about-thesis-project'>
                {length}
              </AboutLengthThesisProject>
              <AboutLocationThesisProject className='about-thesis-project'>
                {location}
              </AboutLocationThesisProject>
              <AboutDescriptionThesisProject className='about-thesis-project'>
                {description}
              </AboutDescriptionThesisProject>
            </AboutThesisProjectAttributes>
          );
        }
      )}
    </AboutThesisProjectContainer>
  );
};

export default AboutThesisProject;
