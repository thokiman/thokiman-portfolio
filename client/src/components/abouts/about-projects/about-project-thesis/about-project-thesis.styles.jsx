import styled, { css } from 'styled-components';

const maxWidthSmartphone = 800;
const whiteSmokeColor = '#e3e1e4';
const lightBlueColor = '#74c6d4';

export const AboutThesisProjectContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 95%;
  margin: 0 auto;
`;

export const AboutThesisProjectTitle = styled.div`
  font-size: 400%;
  padding: 10px;
  color: ${whiteSmokeColor};
  letter-spacing: 0.04em;
  text-align: left;
  display: flex;
  align-items: center;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 18vw;
  }
`;

export const AboutThesisProjectAttributes = styled.div`
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-auto-rows: repeat(5, 1fr);
  gap: 2px 2px;
  grid-template-areas:
    'name-thesis-project name-thesis-project name-thesis-project name-thesis-project name-thesis-project'
    'type-thesis-project type-thesis-project start-end-date-thesis-project start-end-date-thesis-project start-end-date-thesis-project'
    'type-thesis-project type-thesis-project len-thesis-project len-thesis-project len-thesis-project'
    'location-thesis-project location-thesis-project description-thesis-project description-thesis-project description-thesis-project'
    'location-thesis-project location-thesis-project description-thesis-project description-thesis-project description-thesis-project';

  border-bottom: ${lightBlueColor} solid 1px;
  margin: 10px auto;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: fit-content;
  }
`;

const AboutThesisProjectStyles = css`
  width: 100%;
  margin: 10px auto;
  text-align: left;
  display: flex;
  align-items: center;
  text-align: left;
  letter-spacing: 0.04em;
  color: ${whiteSmokeColor};
`;
export const AboutNameThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: name-thesis-project;
  font-size: 220%;
  padding: 10px;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 9vw;
  }
`;

export const AboutTypeThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: type-thesis-project;
  font-size: 140%;
  line-height: 1.1em;
  border-bottom: ${lightBlueColor} solid 1px;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 6vw;
  }
`;

export const AboutLocationThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: location-thesis-project;
  font-size: 140%;
  line-height: 1.1em;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 6vw;
  }
`;
export const AboutStartEndDateThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: start-end-date-thesis-project;
  font-size: 130%;
  border-bottom: ${lightBlueColor} solid 1px;
  line-height: 1.1em;
  letter-spacing: 0.05em;
  padding: 10px 0;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 5.5vw;
  }
`;
export const AboutLengthThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: len-thesis-project;
  font-size: 130%;
  line-height: 1.1em;
  letter-spacing: 0.05em;
  padding: 10px 0;
  border-bottom: ${lightBlueColor} solid 1px;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 5.5vw;
  }
`;
export const AboutDescriptionThesisProject = styled.div`
  ${AboutThesisProjectStyles}
  grid-area: description-thesis-project;
  font-size: 120%;
  line-height: 1.1em;
  letter-spacing: 0.05em;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 4.5vw;
  }
`;

AboutThesisProjectContainer.displayName = 'AboutThesisProjectContainer';
AboutThesisProjectTitle.displayName = 'AboutThesisProjectTitle';
AboutThesisProjectAttributes.displayName = 'AboutThesisProjectAttributes';
AboutNameThesisProject.displayName = 'AboutNameThesisProject';
AboutTypeThesisProject.displayName = 'AboutTypeThesisProject';
AboutLocationThesisProject.displayName = 'AboutLocationThesisProject';
AboutStartEndDateThesisProject.displayName = 'AboutStartEndDateThesisProject';
AboutLengthThesisProject.displayName = 'AboutLengthThesisProject';
AboutDescriptionThesisProject.displayName = 'AboutDescriptionThesisProject';
