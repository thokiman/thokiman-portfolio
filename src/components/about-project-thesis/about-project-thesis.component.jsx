import React from "react";

import {
  AboutDescriptionThesisProject,
  AboutLengthThesisProject,
  AboutLocationThesisProject,
  AboutNameThesisProject,
  AboutStartEndDateThesisProject,
  AboutThesisProjectAttributes,
  AboutThesisProjectContainer,
  AboutThesisProjectTitle,
  AboutTypeThesisProject,
} from "./about-project-thesis.styles";

const AboutThesisProject = ({ thesis }) => {
  return (
    <AboutThesisProjectContainer>
      <AboutThesisProjectTitle>Thesis</AboutThesisProjectTitle>
      {thesis.map(
        ({
          id,
          name,
          type,
          startDate,
          endDate,
          length,
          location,
          description,
        }) => {
          return (
            <AboutThesisProjectAttributes key={id}>
              <AboutNameThesisProject>{name}</AboutNameThesisProject>
              <AboutTypeThesisProject>
                {type.charAt(0).toUpperCase()}
                {type.slice(1, 7)}
                {type.charAt(7).toUpperCase()}
                {type.slice(8)}
              </AboutTypeThesisProject>
              <AboutStartEndDateThesisProject>
                {startDate} - {endDate}
              </AboutStartEndDateThesisProject>
              <AboutLengthThesisProject>{length}</AboutLengthThesisProject>
              <AboutLocationThesisProject>
                {location}
              </AboutLocationThesisProject>
              <AboutDescriptionThesisProject>
                {description}
              </AboutDescriptionThesisProject>
            </AboutThesisProjectAttributes>
          );
        }
      )}
    </AboutThesisProjectContainer>
  );
};

export default AboutThesisProject;
