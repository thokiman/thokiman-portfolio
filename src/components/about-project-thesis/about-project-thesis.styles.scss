.thesis-project-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 95%;
    margin: 0 auto;

    .thesis-project-title {
        font-size: 400%;
        padding: 10px;
        color: #e3e1e4;
        letter-spacing: 0.04em;
        text-align: left;
        display: flex;
        align-items: center;
    }

    .thesis-project {

        width: 100%;
        height: 100%;
        display: grid;
        grid-template-columns: repeat(5, 1fr);
        grid-auto-rows: repeat(5, 1fr);
        gap: 2px 2px;
        grid-template-areas:
            "name-thesis-project name-thesis-project name-thesis-project name-thesis-project name-thesis-project"
            "type-thesis-project type-thesis-project start-end-date-thesis-project start-end-date-thesis-project start-end-date-thesis-project"
            "type-thesis-project type-thesis-project length-thesis-project length-thesis-project length-thesis-project"
            "location-thesis-project location-thesis-project description-thesis-project description-thesis-project description-thesis-project"
            "location-thesis-project location-thesis-project description-thesis-project description-thesis-project description-thesis-project";

        border-bottom: #74c6d4 solid 1px;
        margin: 10px auto;

        .name-thesis-project {
            width: 100%;
            margin: 10px auto;
            grid-area: name-thesis-project;
            font-size: 220%;
            padding: 10px;
            color: #e3e1e4;
            letter-spacing: 0.04em;
            text-align: left;
            display: flex;
            align-items: center;
        }

        .type-thesis-project {
            width: 100%;
            margin: 10px auto;
            grid-area: type-thesis-project;
            font-size: 150%;
            text-align: left;
            display: flex;
            align-items: center;
            text-align: left;
            line-height: 1.1em;
            color: #e3e1e4;
            letter-spacing: 0.04em;
            border-bottom: #74c6d4 solid 1px;

        }

        .location-thesis-project {
            width: 100%;
            margin: 10px auto;
            grid-area: location-thesis-project;
            font-size: 150%;

            display: flex;
            align-items: center;
            text-align: left;
            line-height: 1.1em;
            color: #e3e1e4;
            letter-spacing: 0.04em;

        }

        .start-end-date-thesis-project {
            width: 100%;
            margin: 10px auto;
            grid-area: start-end-date-thesis-project;
            font-size: 130%;
            border-bottom: #74c6d4 solid 1px;
            display: flex;
            align-items: center;
            text-align: left;
            line-height: 1.1em;
            color: #e3e1e4;
            letter-spacing: 0.05em;
            padding: 10px 0;


        }

        .length-thesis-project {
            width: 100%;
            margin: 10px auto;
            grid-area: length-thesis-project;
            font-size: 130%;

            display: flex;
            align-items: center;
            text-align: left;
            line-height: 1.1em;
            color: #e3e1e4;
            letter-spacing: 0.05em;
            padding: 10px 0;
            border-bottom: #74c6d4 solid 1px;

        }



        .description-thesis-project {
            width: 100%;
            margin: 10px auto;
            grid-area: description-thesis-project;
            font-size: 120%;

            display: flex;
            align-items: center;
            text-align: left;
            line-height: 1.1em;
            color: #e3e1e4;
            letter-spacing: 0.05em;
            margin: 10px auto;

        }
    }

}