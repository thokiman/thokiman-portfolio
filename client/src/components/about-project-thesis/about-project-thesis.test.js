import React from "react";
import { shallow } from "enzyme";

import { AboutThesisProject } from "./about-project-thesis.component";

describe("AboutThesisProject component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      thesis: mockArray,
    };
    wrapper = shallow(<AboutThesisProject {...mockProps} />);
  });

  it("should render AboutThesisProject component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
