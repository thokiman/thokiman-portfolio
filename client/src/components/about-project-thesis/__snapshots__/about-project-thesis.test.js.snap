// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutThesisProject component should render AboutThesisProject component 1`] = `ShallowWrapper {}`;
